src/pocketmine/event/block/SignChangeEvent.php

```diff
@@ -79,7 +79,7 @@ public function getLine($index){
 	 * @param string[] $lines
 	 */
 	public function setLines(array $lines){
-		if(count($lines) > 4){
+		if(count($lines) === 4){
```

Comment:
Because you must always have 4 lines in the array or it could lead to undefined behaviour. If you want to set 2 lines you can use setLine().